internal/uploader/events_uploader.go

```diff
@@ -126,6 +105,108 @@ func (e *eventsUploader) sendRequest(req *http.Request) error {
 	return fmt.Errorf("uploading events: upload to %s returned status code %d (body: %s)", req.URL, res.StatusCode, string(body))
 }
 
+// ExtractEvents parses the uploaded content and returns an events per clusterID.
+// content must be a reader of the uploaded binary.
+func ExtractEvents(content io.Reader) ([]eventModels.Events, error) {
```

Comment:
This is a good idea, but perhaps this could go in a separate file? It's not exactly uploading specific

Reply:
I created a new file, let me know if it's fine for you